File: sirb/sirb/report/project_summary_by_irb_unit/project_summary_by_irb_unit.py
# Copyright (c) 2026, Ram and contributors
# For license information, please see license.txt

import frappe


def execute(filters=None):
	columns = [
		{
			"fieldname": "irb_unit",
			"label": "School/Programme",
			"fieldtype": "Data",
		},		
		{
			"fieldname": "project_status",
			"label": "Project Status",
			"fieldtype": "Data",
		},
		{
			"fieldname": "project_count",
			"label": "Project count",
			"fieldtype": "Int",
		},				
	]
	query = f'''select iu.ao_name as irb_unit, count(*) as project_count, p.status as project_status
		from `tabStudent Project Mapping` as sp join `tabIRB Project` as p join `tabIRB Unit` as iu 
		where sp.irb_project = p.name and p.irb_unit = iu.name'''
	if filters and filters["irb_unit"]:
		query += f' and p.irb_unit = \"{filters["irb_unit"]}\" '
	query += ' and sp.status = "Active" group by project_status'
	#print(query)
	results = frappe.db.sql(query, as_dict=1)
	#print(results)
	data = []
	for r in results:
		data.append({"irb_unit": r["irb_unit"], "project_count": r["project_count"], "project_status": r["project_status"]})
	print(data)
	chart = {
		"type": "pie",
		"data": {
			"labels": [x["project_status"] for x in data],
			"datasets": [{"values": [x["project_count"] for x in data]}]
		},
		"isNavigable": True
	}
	return columns, results, "Project count by status for School/Programme", chart

File: sirb/sirb/report/project_summary_by_irb_unit/project_summary_by_irb_unit.js
// Copyright (c) 2026, Ram and contributors
// For license information, please see license.txt
frappe.query_reports["Project Summary By IRB Unit"] = {
	"filters": [
        {
            "fieldname": "irb_unit",
            "label": __("Filter by School/Programme"),
            "fieldtype": "Link",
            "options": "IRB Unit",
            "default": "-- Select --"
        },
	],
	onload: function(report) {
		// Fallback: DOM-level click on the chart SVG slices
		$(report.page.main).on("click", ".frappe-chart path", function() {
			// The selected data is tracked on the chart object
			const chart = report.chart;
			if (chart && chart.state) {
				const index = chart.curActiveSliceIndex;
				const label = chart.data.labels[index];
				frappe.set_route("query-report", "Projects by IRB Unit", {
					"status": label,
					"irb_unit": frappe.query_report.get_filter_value("irb_unit") || ""
				});
			}
		});
	}
}